arch/cortex-m4/src/lib.rs

```diff
@@ -179,6 +191,199 @@ pub unsafe extern "C" fn switch_to_user(
     user_stack as *mut u8
 }
 
+pub unsafe extern "C" fn hard_fault_handler() {
```

Comment:
If this is moving into a common crate, I'd prefer for either there to be some validation that the code overhead of this implementation goes away if `panic` doesn't actually print anything, or have the "stack trace" functionality be feature gated at compile time.

Reply:
I changed Hail's io.rs to:

```rust
/// Panic handler.
#[cfg(not(test))]
#[no_mangle]
#[panic_implementation]
pub unsafe extern "C" fn panic_fmt(pi: &PanicInfo) -> ! {
    // turn off the non panic leds, just in case
    let led_green = &sam4l::gpio::PA[14];
    led_green.enable_output();
    led_green.set();
    let led_blue = &sam4l::gpio::PA[15];
    led_blue.enable_output();
    led_blue.set();

    let led_red = &mut led::LedLow::new(&mut sam4l::gpio::PA[13]);
    // let writer = &mut WRITER;
    // debug::panic(&mut [led_red], writer, pi, &cortexm4::support::nop)
    // Just blink, do not print anything
    debug::panic_blink_forever(&mut [led_red])
}
```
to make sure it would only blink the LEDs and not print anything.

Before:
```
   text	   data	    bss	    dec	    hex	filename
  86018	   4412	  61120	 151550	  24ffe	target/thumbv7em-none-eabi/release/hail
```

After:
```
   text	   data	    bss	    dec	    hex	filename
  51202	   4412	  61120	 116734	  1c7fe	target/thumbv7em-none-eabi/release/hail
```

Also, this function was already in a shared crate, so I'm not sure what has changed (except I guess more boards will be affected). If you wanted to do something differently in the hardfault handler on your board I think you would have the same problem after this PR as you did before.